orangecontrib/text/vectorization/bagofwords.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+from gensim import corpora, matutils
+
+from orangecontrib.text.vectorization.base import BaseVectorizer
+
+
+class CountVectorizer(BaseVectorizer):
+    name = 'Count Vectorizer'
+
+    def __init__(self, binary=False):
+        self.binary = binary
+
+    def transform(self, corpus):
+        corpus = corpus.copy()
+        dic = corpora.Dictionary(corpus.ngrams_iterator(join_with=' '), prune_at=None)
+        X = matutils.corpus2csc(map(dic.doc2bow, corpus.ngrams_iterator(' '))).T
+
+        if self.binary:
+            X[X > 1] = 1
+
+        order = np.argsort([dic[i] for i in range(len(dic))])
```

Comment:
Unfortunately, both solutions won't work.

 `X[:, i]` corresponds to `dic[i]` token, but `[i for i in dic]` does not guarantee increasing order. I have a special test for this.